smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -296,17 +291,18 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
         return sourceAddressToDestinationAddress[tokenAddress] != address(0);
     }
 
-    function verifyNetworkDescriptor(uint256 _networkDescriptor) internal returns(bool) {
+    function _verifyNetworkDescriptor(uint256 _networkDescriptor) private returns(bool) {
         return _networkDescriptor == networkDescriptor;
     }
 
     function _createNewBridgeToken(
-        string calldata symbol,
-        string calldata name,
+        string memory symbol,
+        string memory name,
         address sourceChainTokenAddress,
         uint8 decimals,
-        uint256 _networkDescriptor
-    ) internal returns(address) {
+        uint256 _networkDescriptor,
+        string memory cosmosDenom
+    ) internal returns(address tokenAddress) {
```

Comment:
Seems like we are shadowing a declaration on line 313 with this return value. Instead of doing `returns(address tokenAddress)` we can just do `returns(address)` as that's the style we use across the codebase.

Reply:
Yes, so here's an interesting result:

- If I use `returns(address)`, declare `tokenAddress` in the scope of the function, and then return it 'manually' at the end of the scope, Hardhat tells me that we're spending 1.467.248 gas.

- If I use `returns(address tokenAddress)`, DON'T redeclare `tokenAddress` in the scope of the function, and DON'T return it 'manually', Hardhat tells me that we're spending 1.467.256; it's almost the same amount of gas than before.

- BUT, if I shadow the declaration of tokenAddress by having it both in `returns(address tokenAddress)` AND redeclared in the scope of the function, then Hardhat tells me we're spending 1.429.398 (a 2.65% improvement).

It *is* weird. Should we consider that Hardhat is mistakenly reporting gas usage? I don't understand why we're saving gas by doing this convoluted thing, it sure looks like a mistake.